LocalParks.API/Controllers/ApiParkEventsController.cs

```diff
@@ -0,0 +1,145 @@
+﻿using LocalParks.Core.Contracts;
+using LocalParks.Core.Models;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Globalization;
+using System.Threading.Tasks;
+
+namespace LocalParks.API.Controllers
+{
+    [ApiController]
+    [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme)]
+    public class ParkEventsController : ControllerBase
+    {
+        private readonly ILogger<ParkEventsController> _logger;
+        private readonly IParkEventsService _service;
+
+        public ParkEventsController(ILogger<ParkEventsController> logger, IParkEventsService service)
+        {
+            _logger = logger;
+            _service = service;
+        }
+
+        [HttpGet("api/[controller]")]
+        public async Task<ActionResult<ParkEventModel[]>> GetAllParkEvents()
+        {
+            _logger.LogInformation("API GET request: All Events");
+
+            try
+            {
+                var results = await _service.GetAllParkEventModelsAsync();
+
+                if (results == null) return NoContent();
+
+                return Ok(results);
+            }
+            catch (Exception ex)
```

Comment:
You can register exception handler/strategies in your startup class so you don't have to add a try catch in all your controllers apis ... 

These might help.
https://weblog.west-wind.com/posts/2016/oct/16/error-handling-and-exceptionfilter-dependency-injection-for-aspnet-core-apis

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-5.0

Reply:
Ah ok, thanks, James. I'll move this and the logging into a handler.